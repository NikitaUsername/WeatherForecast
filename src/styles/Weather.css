.mainInfo {
    display: inline-block;
    width: 100%;
}

.mainWeatherInfo {
    width: 100%;
}

.mainWeatherInfo__temp {
    margin-top: 1em;
    font-size: 2.8rem;
}

.mainWeatherInfo__pic {
    margin-top: 1.25rem;
    height: 7.5rem;
}

.weatherElement__value {
    font-size: 1.25rem;
    margin-left: auto;
    margin-right: 1em;
    font-weight: bold;
    color: #757575;
    text-align: right;
}

.weatherElement {
    padding-top: 0.75rem;
    background-color: lightgrey;
    border-radius: 0.65rem;
    margin-bottom: 0.3rem;
}