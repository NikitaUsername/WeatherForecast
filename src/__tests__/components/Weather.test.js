import Weather from '../../components/Weather'
import React from 'react'
import renderer from 'react-test-renderer'

describe('Weather component', () => {
    test('matches the snapshot with city', () => {

        const info = 
            {
                name: 'Moscow',
                weather: [
                    { id: 500, main: "Rain", description: "light rain", icon: "10n" }
                ],
                main: {
                    temp: 20.46,
                    feels_like: 19.46,
                    temp_min: 19.44,
                    temp_max: 21.67,
                    pressure: 1021,
                    humidity: 83
                },
                wind: {
                    speed: 5.1,
                    deg: 40
                },
                clouds: {
                    all:90
                }
            };
        
        const tree = renderer
            .create (<Weather weatherInfo={info}/>).toJSON();
            expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot with loader', () => {

        const info = undefined;
        
        const tree = renderer
            .create (<Weather weatherInfo={info}/>).toJSON();
            expect(tree).toMatchSnapshot();
    });
});