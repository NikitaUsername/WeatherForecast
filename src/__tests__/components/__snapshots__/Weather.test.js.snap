// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component matches the snapshot with city 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6"
    >
      <h2
        id="cityname"
      >
        Moscow
      </h2>
      <div
        className="row"
      >
        <img
          id="pic"
          src="http://openweathermap.org/img/wn/10n@2x.png"
        />
        <p
          id="temp"
        >
           
          20.46
           ºC 
        </p>
      </div>
    </div>
    <div
      className="col-lg-6"
    >
      <div
        className="row"
         
      >
        <p>
          Погода
        </p>
        <div
          id="val"
        >
           
          light rain
           
        </div>
      </div>
      <div
        className="row"
         
      >
        <p>
          Давление
        </p>
        <div
          id="val"
        >
           
          1021
           hPA
        </div>
      </div>
      <div
        className="row"
         
      >
        <p>
          Скорость ветра
        </p>
        <div
          id="val"
        >
           
          5.1
           m/s
        </div>
      </div>
      <div
        className="row"
         
      >
        <p>
          Облачность
        </p>
        <div
          id="val"
        >
           
          90
           %
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Weather component matches the snapshot with loader 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <h2
        id="waiting"
      >
        Подождите, данные загружаются
      </h2>
    </div>
    <div
      className="col-lg-12 "
    >
      <div
        className="center-block"
      >
        <div
          className="css-236bqq"
        />
      </div>
    </div>
  </div>
</div>
`;
